### ISSUE TITLE
MSLError: Not permitted to switch to the requested user profile

### ISSUE THREAD

**Issue by michkot** [CastagnaIT/plugin.video.netflix#1402]
### Netflix add-on version

1.18.9+matrix.1

### Operative systems used

Android

### Kodi version used

Kodi 19 (Matrix)

### Description of the bug

When trying to play any show my with "my" profile, I got this error:
MSLError: Not permitted to switch to the requested user profile

Later I found out, that using the default/owner account works (but I don't wanna use that for me / on this device / not now). Switching back to "my" profile reproduced the error.

BTW: I see a bunch of "signatures" profile GUIDs in the log, not so sure I would call this non-personal... (didn't do any filtering though, as requested)

### Debug log - mandatory

https://privatebin.net/?89b0119bccd8ef75#5ACi5RiPEJdYVHqbmeXmD5LncggzwZChHbiwiLNKJDgi

### Additional context

This in on Amazon Fire TV Stick 4k max and the profile works completely find on the build-in app and also in browser on other devices.

**Comment by CastagnaIT** [comment]
there is no personal data in the log, at least not from this addon
the profile GUID are useless without access to your account and without having deeper knowledgle in how works website API
and in any case are needed only to switch account and minor things so are not so importants

> 2022-09-06 23:40:37.690 T:29493    INFO <general>: [plugin.video.netflix (3)] The debug logging is set as DISABLED

the log is disabled follow instruction on Readme github

**Comment by michkot** [comment]
Okey, I will try again with restarting the app - I clearly see the DEBUG
messages that were not present before I switched the Debug switch in
settings.

**Comment by karnadii** [comment]
can confirm, I have the same error when playing on profile other than profile 1.

**Comment by chathurach** [comment]
I'm getting the same error.

**Comment by CastagnaIT** [comment]
people write here "i also have the error", it is not helpful to solve the problem add only garbage to the issue thread
nobody has posted debugging logs or and/or other useful information

atm i am currently unable to replicate this problem in any way so it is impossible for me to investigate
probably atm is a change limited only to targeted users or specifics settings

**Comment by chathurach** [comment]
Sorry for the previous comment. Here is the log. Hope it helps. Thank you for the great addon.
[https://paste.kodi.tv/papajohewu.kodi](url)

**Comment by CastagnaIT** [comment]
thank you i think i have found something in the website
i will need to investigate better in next days

**Comment by CastagnaIT** [comment]
so found that this new endpoint 
`https://www.netflix.com/api/shakti/mre/profiles/switch`
is called to switch profile but seems used only for nfsession side for is not useful for MSL side api

the problem from the logs is that the MSL user-authentication scheme `SWITCH_PROFILE` (https://github.com/CastagnaIT/plugin.video.netflix/wiki/Dev.-basic-info#msl-apis) used to switch profile on MSL side
for some reason appears to not works

currently we attach the auth data to MSL request as:

```json
"userauthdata": {
      "scheme": "SWITCH_PROFILE",
      "authdata": {
            "useridtoken": token_data,
            "profileguid": selected_user_profile_guid
}
```

instead the website still use `NETFLIXID` MSL user-authentication scheme

```json
    "userauthdata": {
        "scheme": "NETFLIXID",
        "authdata": {...}
    }
```

this last one, its a problem for us, because it cannot be used on Android devices
maybe it's worth trying again

**Comment by chathurach** [comment]
This error pops when I try to watch a video with the owner profile- "Sorry, your account can't be used on this device"
log - https://paste.kodi.tv/aserelejoc.kodi
Not sure if it is related to this or something else.

**Comment by CastagnaIT** [comment]
@chathurach not concern this Issue
more likely you are using a banned device or you have an type of account cannot be used here

**Comment by CastagnaIT** [comment]
from 2 day ago i am finally able to reproduce this error
so i tried set NETFLIXID auth and if i dont have do wrong things seem to works now also on android!

i have done a first test build, in theory should works if updated from a previous addon installation
(so not tested if works from addon installed from scratch)
[plugin.video.netflix_1.18.9+matrix.1_20220924_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/9639126/plugin.video.netflix_1.18.9%2Bmatrix.1_20220924_t1.zip)

if someone can update addon to his android device and let me know if works,
**important thing** if the video playback works please check that the "continue watching" is updated to the right netflix profile
(play at least 3 minutes of video)

**Comment by michkot** [comment]
TL;DR
tried the updated version, and it works correctly including "continue watching"

- updated the plugin from the attached zip
- made sure the profile is still set to non-owner
- picked something from continue watching (I repeated this process a few times)
- - watched for 3 - 10 minutes, did some time skips too
- - stopped playback
- - exited the kodi app
- a) switched to the netflix app which was likely still running in the background, the continue watching position did not change
- b) restarted the whole device (tried before without), switched to the netflix app, the episode was marked as viewed (because I time skipped to the end in the meantime)
- c) switched to the netflix app, switched profiles to different one and back, the watching position was correctly up to date with what I watched in kodi plugin

**Comment by CastagnaIT** [comment]
thanks for the detailed answer

> a) switched to the netflix app which was likely still running in the background, the continue watching position did not change

yes if you have original netflix app active in background take some time get "watched status" updates, otherwise force close the app, and reopen it

> the episode was marked as viewed (well I ended a playback in different position, but I time skipped to the end by accident

if you skip near the end (depends from video) could be set as full watched, this is a regular behaviour of netflix itself

seem to works fine i think i can finish the changes

**Comment by CastagnaIT** [comment]
Ok tested also with a fresh addon install, so this is the full fix test build:
[plugin.video.netflix_1.18.9+matrix.1_20220925_t2.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/9640468/plugin.video.netflix_1.18.9%2Bmatrix.1_20220925_t2.zip)
